src/main/java/bio/terra/service/common/CreateAzureContainerStep.java

```diff
@@ -0,0 +1,86 @@
+package bio.terra.service.common;
+
+import static bio.terra.service.common.CommonMapKeys.SHOULD_PERFORM_CONTAINER_ROLLBACK;
+
+import bio.terra.model.BillingProfileModel;
+import bio.terra.service.profile.flight.ProfileMapKeys;
+import bio.terra.service.resourcemanagement.ResourceService;
+import bio.terra.service.resourcemanagement.azure.AzureContainerPdao;
+import bio.terra.service.resourcemanagement.azure.AzureStorageAccountResource;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import com.azure.storage.blob.BlobContainerClient;
+
+/**
+ * Extend this class to ensure that a created storage container gets created in a way that can be
+ * cleaned up if the flight fails
+ */
+public abstract class CreateAzureContainerStep implements Step {
+  protected final ResourceService resourceService;
+  protected final AzureContainerPdao azureContainerPdao;
+
+  public CreateAzureContainerStep(
```

Comment:
I think this can be `protected`
```suggestion
  protected CreateAzureContainerStep(
```

Reply:
good call!